Pre RM-6801: Add ScreenshotTransformation Tests, Merge TransformationHelper & ScreenshotContext

iam-mholle opened the pull request.

Stefan-Ilic commented on Remotion/Web/Development.WebTesting.IntegrationTests/ScreenshotTest.cs:
```diff
@@ -724,5 +748,66 @@ private class DerivedScreenshotWebTreeViewNodeControlObject : ScreenshotWebTreeV
+                context.ResolvedElement.ElementBounds.Width,
+                context.ResolvedElement.ElementBounds.Height - _pixelToCutOff));
+
+        return context.CloneWith(resolvedElement: resolvedElement);
```
Missing whitespace

Stefan-Ilic commented on Remotion/Web/Development.WebTesting.IntegrationTests/ScreenshotTest.cs:
```diff
@@ -724,5 +748,66 @@ private class DerivedScreenshotWebTreeViewNodeControlObject : ScreenshotWebTreeV
+      private int Width { get; }
+      private int Height { get; }
+      private Brush Brush { get; }
+      public Point Offset { get; }
```
Can be made `private`

Stefan-Ilic commented on Remotion/Web/Development.WebTesting.IntegrationTests/ScreenshotTest.cs:
```diff
@@ -724,5 +748,66 @@ private class DerivedScreenshotWebTreeViewNodeControlObject : ScreenshotWebTreeV
+      }
+    }
+
+    private class RemoveYScreenshotTransformation<T> : IScreenshotTransformation<T>
```
The name might indicate that it can remove stuff across the entire Y axis. Perhaps rename the class to `RemovePixelsFromBottomScreenshotTransformation` or something like that

Stefan-Ilic commented on Remotion/Web/Development.WebTesting/ScreenshotCreation/ScreenshotTransformationContext.cs:
```diff
@@ -25,31 +25,77 @@ namespace Remotion.Web.Development.WebTesting.ScreenshotCreation
-  public class ScreenshotTransformationContext<T>
+  public class ScreenshotTransformationContext<T> : IDisposable
   {
+    public static ScreenshotTransformationContext<T> CreateForTransformation (
```
Missing documentation

Stefan-Ilic commented on Remotion/Web/Development.WebTesting/ScreenshotCreation/ScreenshotTransformationContext.cs:
```diff
@@ -25,31 +25,77 @@ namespace Remotion.Web.Development.WebTesting.ScreenshotCreation
+                resolvedElement:
+                resolvedElement.CloneWith (elementBounds: Rectangle.Intersect (parentBounds.Value, context.ResolvedElement.ElementBounds)));
+      }
+      else
```
This else-block is unnecessary. Did you leave it here intentionally?

Stefan-Ilic commented on Remotion/Web/Development.WebTesting/ScreenshotCreation/ScreenshotTransformationContext.cs:
```diff
@@ -25,31 +25,77 @@ namespace Remotion.Web.Development.WebTesting.ScreenshotCreation
         [NotNull] IScreenshotElementResolver<T> resolver,
         [NotNull] T target,
         [NotNull] ResolvedScreenshotElement resolvedElement)
+        : this (manipulation, graphics, resolver, target, resolvedElement, null)
```
Does it make sense to initialize a `ScreenshotTransformationContext` without a `ScreenshotTransformation`? Could we perhaps only leave the ctor that requires a transformation and leave out this ctor and the factory method?

iam-mholle commented:
- Added `ScreenshotTransformation` tests in https://github.com/re-motion/Trunk-SVN2Git/commit/21722ee29da584dac051b31b6aa452cb1dbdafc6
- Added issue [RM-7222](https://re-motion.atlassian.net/browse/RM-7222) for the refactoring of `ScreenshotTransformationContext` and `TransformationHelper`.
